### PR TITLE
refactor: Upgrade GraphQL dependencies

### LINKED ISSUE #7916: Upgrade GraphQL dependencies
### New Feature / Enhancement Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Current Limitation
GraphQL dependencies have updates available, which require code changes to get merged. Tests for MongoDB are failing.

### PR DESCRIPTION
**Pull request by Moumouls**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Issue Description

Update graphql deps

Related issue: #7916
Closes #7916

### Approach

Update all packages to latest version

### TODOs before merging

- [x] Add tests
- [x] Add changes to documentation (guides, repository pages, in-code descriptions)
- [x] Add [security check](https://github.com/parse-community/parse-server/blob/master/CONTRIBUTING.md#security-checks)
- [x] Add new Parse Error codes to Parse JS SDK 
- [x] A changelog entry is created automatically using the pull request title (do not manually add a changelog entry)

### CODE REVIEW

**Comment by Moumouls** [comment]
I'm not in favor of merging this PR currently since it introduces a possible breaking change with the async apply graphql server.

This PR should be merged after the switch to Yoga: https://github.com/parse-community/parse-server/pull/7967

**Comment by Moumouls** [comment]
@mtrezza  we are good to go here 🚀

**Comment by mtrezza** [comment]
Could you just give this PR a proper title? Remember, it will be used as the changelog entry and that is impacting the DX.

**Comment by Moumouls** [comment]
done @mtrezza

File: src/GraphQL/ParseGraphQLSchema.js [hunk 1]
```diff
@@ -203,7 +203,6 @@ class ParseGraphQLSchema {
 
     if (this.graphQLCustomTypeDefs) {
       schemaDirectives.load(this);
-
       if (typeof this.graphQLCustomTypeDefs.getTypeMap === 'function') {
         // In following code we use underscore attr to avoid js var un ref
```
Comment (mtrezza) [inline src/GraphQL/ParseGraphQLSchema.js]: Could you clarify what "js var un ref" means?

**Review by mtrezza — CHANGES_REQUESTED** [review]

File: src/GraphQL/ParseGraphQLSchema.js (hunk 1 above)
Comment (Moumouls) [reply src/GraphQL/ParseGraphQLSchema.js]: i updated the comment

**Comment by Moumouls** [comment]
We should be good to go here @mtrezza 🚀

**Review by davimacedo — APPROVED** [review]
it looks good to me

**Review by dblythy — APPROVED** [review]

**Comment by Moumouls** [comment]
@mtrezza could you merge this one ?

**Comment by mtrezza** [comment]
Rerunning failing CI

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.3.0-alpha.16](https://github.com/parse-community/parse-server/releases/tag/5.3.0-alpha.16)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.3.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/5.3.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.3.0](https://github.com/parse-community/parse-server/releases/tag/5.3.0)

**Pull request merged**